rts/motoko-rts/src/bigint.rs

```diff
@@ -523,3 +564,48 @@ pub unsafe extern "C" fn bigint_sleb128_decode(buf: *mut Buf) -> Value {
 
     persist_bigint(i)
 }
+
+#[cfg(feature = "ic")]
+#[no_mangle]
+pub unsafe extern "C" fn bigint_sleb128_decode_word64(
+    mut leb: i64,
+    bits: u64,
+    buf: *mut Buf,
+) -> Value {
+    let mut acc = leb & 0b111_1111;
+    if bits < 8 {
+        buf.advance(1);
+        return Value::from_signed_scalar((acc as i32) << 25 >> 25);
```

Comment:
These are ugly. one probably could just copy into the upper 7 bits all the time and then just shift once.

Reply:
9f06cca12 builds up the result from the top-bits perspective. It has a simpler sign-extension regime, but still has the same number of shifts.